### DISCUSSION TITLE
Javascript Error while Rendering

### DISCUSSION THREAD

**Discussion by arnavdas88** [idom-team/idom#1153] [Question] [answered]
I am trying to build a component library on top of [reactpy](https://github.com/reactive-python/reactpy) for [Nivo Charts](https://github.com/plouc/nivo). 

After following the [documentation](https://reactpy.dev/docs/guides/escape-hatches/javascript-components.html#dynamically-loaded-components), I have imported the library nivo charts into python and imported a chart like this:

```python
from reactpy import component, run, web

nivo_module = web.module_from_template(
    "react",
    "@nivo/core",
    fallback="⌛",
)
nivo_bar_module = web.module_from_template(
    "react",
    "@nivo/bar",
    fallback="⌛",
)

ResponsiveBar = web.export(nivo_bar_module, "ResponsiveBar")
```

Once imported, I am using the exported object `ResponsiveBar` as below:
```python
@component
def HelloWorld():
    return html.section(
        ResponsiveBar({"data":data, "keys": [ 'hot dog', 'burger', 'sandwich', 'kebab', 'fries', 'donut'], "indexBy": "country", "role": "application"})
    )


app = FastAPI()
configure(app, HelloWorld)
```

I am trying to replicate the example showcased in [Nivo Playground](https://nivo.rocks/bar/)

When running the uvicorn server, while rendering, I am getting a javascript error `Uncaught (in promise) SyntaxError: ambiguous indirect export: isString`

Q.) Is this a problem from the Nivo charts?
Q.) Is it because of some conversion in `module_from_template`?
Q.) Is there something that I am doing wrong?
Q.) ~~Or, Is this normal?~~

**Comment by Archmonger** [comment, marked as answer]
`module_from_template` is deprecated due to instability and will be removed in the next major release.
For now, we suggest using the custom components API instead.

**Reply by arnavdas88** [reply]
After using the `Custom Components API`, the code works properly, without errors, but every time I am using the exported component,

```python
ResponsiveBar({"data":data, "keys": [ 'hot dog', 'burger', 'sandwich', 'kebab', 'fries', 'donut'], "indexBy": "country", "role": "application"})
```

It renders inside a `<div></div>`. Containing the component inside an `html.div` just add another div.
```python
# My Code
@component
def HelloWorld():
    return html.div(
        {"style": "height:500px;"},
        ResponsiveBar({"data":data, "keys": [ 'hot dog', 'burger', 'sandwich', 'kebab', 'fries', 'donut'], "indexBy": "country", "role": "application"})
    )

app = FastAPI()
configure(app, HelloWorld)

```
```HTML
<!-- Rendered HTML -->
<div style="height: 500px;">
    <div>
        <div style="width: 100%; height: 100%;"></div>
    </div>
</div>
```

I am unable to control anything regarding the middle `<div>`. Although, it can be controlled, if i set style for all div globally, but is there any other way to control it? like using some parameters/arguments? Does all custom components get wrapped by div?

**Reply by Archmonger** [reply]
@rmorshea Can you chime in?
